plugins/rest/rest_auth.go

```diff
@@ -149,10 +257,28 @@ func (ap *oauth2ClientCredentialsAuthPlugin) NewClient(c Config) (*http.Client,
 	return DefaultRoundTripperClient(t, *c.ResponseHeaderTimeoutSeconds), nil
 }
 
-// requestToken tries to obtain an access token using the client credentials flow
+// requestToken tries to obtain an access token using either the client credentials flow
 // https://tools.ietf.org/html/rfc6749#section-4.4
+// or the JWT authorization grant
+// https://tools.ietf.org/html/rfc7523
 func (ap *oauth2ClientCredentialsAuthPlugin) requestToken() (*oauth2Token, error) {
-	body := url.Values{"grant_type": []string{"client_credentials"}}
+	var body url.Values
+	if ap.GrantType == "jwt_bearer" {
+		authJwt, err := ap.createAuthJwt(ap.Claims, ap.signingKeyParsed)
```

Comment:
[nit] I think the golang style guide would want the acronym to be spelled with capital letters here,
```suggestion
		authJWT, err := ap.createAuthJWT(ap.Claims, ap.signingKeyParsed)
```
...but it's really not important. 😉

Reply:
Sure, I'll change 👍